src/main/java/info/u_team/music_player/dependency/classloader/DependencyClassLoader.java

```diff
@@ -1,12 +1,19 @@
 package info.u_team.music_player.dependency.classloader;
 
+import cpw.mods.cl.ModularURLHandler;
+
 import java.net.URL;
 import java.net.URLClassLoader;
 
 public class DependencyClassLoader extends URLClassLoader {
 	
 	static {
 		ClassLoader.registerAsParallelCapable();
+		/*
+		 Call ServiceLoader.load(ModuleLayer, IURLProvider.class) and put to private Map<String, IURLProvider> handlers
+		 (See resources/META-INF/services/cpw.mods.cl.ModularURLHandler$IURLProvider and cpw.mods.cl.ModularURLHandler#initFrom)
+		 */
+		ModularURLHandler.initFrom(DependencyClassLoader.class.getModule().getLayer());
```

Comment:
Please confirm that after the initFrom, the ModularURLHandler still works with the union urls defined here: https://github.com/McModLauncher/securejarhandler/blob/main/src/main/java/cpw/mods/cl/UnionURLStreamHandler.java

Reply:
Sorry for the late reply!
I tried this by writing:
```java
 public static void checkHandlers() {
        try {
            final var field = ModularURLHandler.class.getDeclaredField("handlers");
            final Map<String, IURLProvider> handlers = UnsafeHacks.getField(field, ModularURLHandler.INSTANCE);
            for (Map.Entry<String, IURLProvider> entry : handlers.entrySet()) {
                LOGGER.info("Protocol: {}, Provider: {}", entry.getKey(), entry.getValue());
            }
        } catch (Exception e) {
            e.printStackTrace();
        }
    }
}
```
And here's how I implemented it:
```java
public static void load() {
        LOGGER.info(MARKER_LOAD, "Load dependencies");

        checkHandlers(); // <- here

        //setupURLStreamHack();

        final String devPath = System.getProperty("musicplayer.dev");
        if (devPath != null) {
            findJarFilesInDev(Paths.get(devPath, "musicplayer-lavaplayer/build/libs"), path -> addToMusicPlayerDependencies(pathToUrl().apply(path)));
            findJarFilesInDev(Paths.get(devPath, "musicplayer-lavaplayer/build/dependencies"), path -> addToMusicPlayerDependencies(pathToUrl().apply(path)));
        } else {
            findJarFilesInJar("dependencies", path -> addToMusicPlayerDependencies(createInternalURL(path)));
        }

        LOGGER.info(MARKER_LOAD, "Finished loading dependencies");
    }
```

The log output showed something like this:
```java
[10:36:01] [modloading-worker-0/INFO]: Protocol: union, Provider: cpw.mods.cl.UnionURLStreamHandler@d0d3e4f
[10:36:01] [modloading-worker-0/INFO]: Protocol: jarlookup, Provider: info.u_team.music_player.dependency.urlprovider. DependencyURLProvider@76582b91.
```

I checked and the UnionURLStreamHandler is assigned in module-info.java 
(https://github.com/McModLauncher/securejarhandler/blob/main/src/main/java/module-info.java)
```java
uses cpw.mods.cl.ModularURLHandler.IURLProvider;// (line 13)
provides ModularURLHandler.IURLProvider with UnionURLStreamHandler;// (line 14)
``` 
Will this load on its own? Maybe.

After that, when I assigned the IURLProvider to other mods, the minecraft log output looked like this when I tried this too:
```java
[11:02:58] [modloading-worker-0/INFO]: Protocol: test, Provider: com.yukkuritaku.touhoumod.TestURLProvider@6d810356
[11:02:58] [modloading-worker-0/INFO]: Protocol: union, Provider: cpw.mods.cl.UnionURLStreamHandler@50f1eaef
[11:02:58] [modloading-worker-0/INFO]: Protocol: jarlookup, Provider: info.u_team.music_player.dependency.urlprovider.DependencyURLProvider@4cf53e3b
```
(However, I haven't tried it in my development environment...)
So I think it's okay! (But I haven't researched the spi specs much yet myself, so I could be wrong...)